lib/src/layer.dart

```diff
@@ -1,53 +1,54 @@
-part of tiled;
+import 'package:xml/xml.dart';
+import '../tiled.dart';
+import 'tile_map_parser.dart';
+import 'tile_map.dart';
+import 'tile.dart';
+import 'node_dsl.dart';
+import 'flips.dart';
 
 class Layer {
   static const int FLIPPED_HORIZONTALLY_FLAG = 0x80000000;
   static const int FLIPPED_VERTICALLY_FLAG = 0x40000000;
   static const int FLIPPED_DIAGONALLY_FLAG = 0x20000000;
 
-  String name;
-  int width;
-  int height;
-  bool visible;
+  String name = "";
+  int width = 0;
+  int height = 0;
+  bool visible = true;
 
-  Map<String, dynamic> properties = {};
+  Map<String?, dynamic> properties = {};
 
-  TileMap map;
-  List<List<int>> tileMatrix;
-  List<List<Flips>> tileFlips;
+  TileMap? map;
+  late List<List<int>> tileMatrix;
+  late List<List<Flips>> tileFlips;
 
-  List<List<Tile>> _tiles;
-  List<List<Tile>> get tiles {
+  List<List<Tile>>? _tiles;
+  List<List<Tile>>? get tiles {
```

Comment:
This shouldn't need the `?` right?

Reply:
In this case I think it is a complex data structure, an it comes form somwhere and there is no meaningful simple initial value.

Reply:
On this getter we know that it is initialized, so you can just return `_tiles!`

Reply:
Yes, the `?` could be removed for the getter, but what you want to return instead if `_tiles` is `null`?

Reply:
It can't be null in the getter, since it does `_recalculateTiles()` if it is null.

Reply:
...it should be `_tiles!`